alphagov/cf-terraform
# Vanilla cf prototype diego

**Note:** This is against `vanilla_cf_prototype` not master. We're PR'ing it here so the commits can be tested as the new manifests repo is not yet deployable. Once this is merged these commits need extracting and reapplying to the new manifests repository.

This is in parallel to our existing CF deployment using DEA's and users can use the `cf disable-diego` command to switch an individual app back to running on the legacy infrastructure.
## Future work
- SSL has been disabled for all internal Diego infrastructure communication, mainly BBS. This should be addressed in a future story along with other random credential generation.
- `cf ssh` to containers does not work yet due to a [bug in Diego](https://github.com/cloudfoundry/uaa/issues/223) that will be fixed in future releases. Arguably we shouldn't enable it at all.
- ELB healthchecks have been moved to port 22 as both routers and access machines listen on it. ELB healthchecks can only be configured to ping one port, and `cf ssh` requires port 2222. This means that we're not effectively healthchecking our routers or access machines, and this decision should be revisited in a future story.
- All the old DEA infrastructure should be removed from the manifests once we're happy that Diego functions as expected, then `cc.users_can_select_backend` should be set to false.
## How to test this PR

`make aws DEPLOY_ENV=iamtestingdiego`, then try deploying an application to the CF instance.

When deploying the application you will notice that the buildpack installation process is different and starts with "Creating container", then waffles about downloading buildpacks for a while:

```
Starting app spring-music in org GDS / space jim as admin...
Creating container
Successfully created container
Downloading app package...
Downloaded app package (21.1M)
No buildpack specified; fetching standard buildpacks to detect and build your application.
Downloading buildpacks (staticfile_buildpack, java_buildpack, ruby_buildpack, nodejs_buildpack, go_buildpack, python_buildpack, php_buildpack, binary_buil
dpack)...
Downloading nodejs_buildpack...
Downloading ruby_buildpack...
Downloading go_buildpack...
Downloading staticfile_buildpack...
Downloaded staticfile_buildpack
Downloading python_buildpack...
Downloading php_buildpack...
Downloaded nodejs_buildpack
Downloading binary_buildpack...
Downloaded binary_buildpack
Downloading java_buildpack...
Downloaded java_buildpack
Downloaded python_buildpack
Downloaded ruby_buildpack
Downloaded php_buildpack
Downloaded go_buildpack
Downloaded buildpacks
Staging...
```
## Who can review this PR

Anybody but @jimconner and @jonty, who disliked this less than concourse.
